src/utils/combineTodoAndUser.ts

```diff
@@ -0,0 +1,34 @@
+import { Todo } from '../types/Todo';
+import { TodoWithUser } from '../types/TodoWithUser';
+import { User } from '../types/User';
+
+export const combineTodoAndUser = (
+  todos: Todo[], users: User[],
+): Array<TodoWithUser> => {
+  const combinedData = todos.map((todo: Todo) => {
+    const userObj: User | null
+    = users.find((user: User) => user.id === todo.userId) || null;
+
+    if (userObj !== null) {
+      return {
+        id: todo.id,
+        title: todo.title,
+        completed: todo.completed,
+        userId: todo.userId,
+        user: {
+          id: userObj.id,
+          name: userObj.name,
+          username: userObj.username,
+          email: userObj.email,
```

Comment:
It is necessary to create a separate type for the user and use it here for better readability